### PR TITLE
[fix](load) start transaction before we need it (#8819)

### LINKED ISSUE #8819: [Bug] Failed to execute SparkLoadJob
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master branch

### What's Wrong?

Hi, currently I am trying Spark Load Job，but ended up with error msg like this every time:  txn does not exist. id: 0

+-------+--------------+-----------+-------------------+-------+---------+-------------------------------------------------------+----------------------------------------------------------------------------------+---------------------+--------------+---------------+---------------+---------------------+------+-----------------------------------------------------------------------------------------------------------------------+---------------+--------------+
| JobId | Label        | State     | Progress          | Type  | EtlInfo | TaskInfo                                              | ErrorMsg                                                                         | CreateTime          | EtlStartTime | EtlFinishTime | LoadStartTime | LoadFinishTime      | URL  | JobDetails                                                                                                            | TransactionId | ErrorTablets |
+-------+--------------+-----------+-------------------+-------+---------+-------------------------------------------------------+----------------------------------------------------------------------------------+---------------------+--------------+---------------+---------------+---------------------+------+-----------------------------------------------------------------------------------------------------------------------+---------------+--------------+
| 16006 | doris_goods  | CANCELLED | ETL:N/A; LOAD:N/A | SPARK | NULL    | cluster:spark0; timeout(s):3600; max_filter_ratio:0.0 | type:ETL_SUBMIT_FAIL; msg:errCode = 2, detailMessage = txn does not exist. id: 0 | 2022-04-01 19:41:39 | NULL         | NULL          | NULL          | 2022-04-01 19:41:44 | NULL | {"Unfinished backends":{},"ScannedRows":0,"TaskNumber":0,"LoadBytes":0,"All backends":{},"FileNumber":0,"FileSize":0} | 0             | {}           |

### What You Expected?

After going through the code，the Error generated at the following code inside this fuction SparkLoadPendingTask#createEtlJobConf

`// add table indexes to transaction state`
`TransactionState txnState = Catalog.getCurrentGlobalTransactionMgr()`
                            `.getTransactionState(dbId, transactionId);`
`if (txnState == null) {`
        `throw new LoadException("txn does not exist. id: " + transactionId);`
`}`

Seems that the transaction has not begun（transactionId is 0 as shown in error message). The transaction begins lately when SparkLoadPendingTask start to execute. 

I wonder if there is something wrong in my test procedure, but I did follow the guide in official document. 

Thanks for your help .

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhengshengjun**
# Proposed changes

Issue Number: close #8819 

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/load/loadv2/SparkLoadPendingTask.java [hunk 1]
```diff
@@ -145,6 +142,13 @@ private void createEtlJobConf() throws LoadException {
             throw new LoadException(e.getMessage());
         }
 
+        try {
+            ((SparkLoadJob) callback).beginTxn();
+        } catch (UserException e) {
+            LOG.error("failed to begin transaction", e);
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/load/loadv2/SparkLoadPendingTask.java:148]:
```suggestion
            LOG.warn("failed to begin transaction for spark load job {}", this.loadJobId, e);
```

**Comment by wangbo** [comment]
How about calling beginTnx() in ```SparkLoadJob.unprotectedExecuteJob```, it seems more reasonable.
Also closer to the original version.
```SparkLoadJob``` is responsible for business management, ```SparkLoadPendingTask``` is responsible for task execution.

**Comment by zhengshengjun** [comment]
> How about calling beginTnx() in `SparkLoadJob.unprotectedExecuteJob`, it seems more reasonable. Also closer to the original version. `SparkLoadJob` is responsible for transaction management, `SparkLoadPendingTask` is responsible for task execution.

Good suggestion! I had thought to start transaction where it is now stared （in `SparkLoadPendingTask`). But it's more reasonable to manage a job level transaction in `SparkLoadJob` as you mentioned, fixed.

**Review by wangbo — APPROVED** [review]
LGTM

**Pull request merged**